### PR TITLE
[dotnet] Correct assertion of struct type in internal tests

### LINKED ISSUE #14852: [dotnet] Future-proof for NUnit 4.x upgrade
### Feature and motivation

With the release of NUnit 4.x, a few major breaking changes have been added. We want our test suite to be able to be prepared for NUnit 4, even if we're not going to upgrade for a while yet.

- [x] **Classic Assert** #14853 #14870
 The "classic assert" model has been depreciated. In order to continue using them, the tests would need to jump through some hoops.

  A better solution would be to migrate to modern NUnit assertions. The `NUnit.Analyzers` nuget package has useful features to accomplish this. Examples include:
  - `Assert.AreEqual("Testing Alerts", driver.Title)` -> `Assert.That(driver.Title, Is.EqualTo("Testing Alerts"))`
  - `Assert.AreEqual(1, allWindows.Count)` -> `Assert.That(allWindows, Has.One.Items)`

- [x] String format messages #14878
  NUnit 3.x has methods such as `Assert.That(x, Is.EqualTo(y), "Value {0} is {1}", x, y)`. These overloads have been removed, in order to make room for caller argument expression parameters. We will have to go through and turn these all into interpolated strings. Same For `Assert.Ignore(string message, string[] args)`.

- [x] The handful of uses of `StringAssert` or `CollectionAssert` should be changed to `Assert.That`. #14870

### PR DESCRIPTION
**Pull request by RenderMichael**
### **User description**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Finishes future-proofing for a potential NUnit 4 upgrade. With this, I upgraded locally to NUnit 4.x and the tests ran and passed.

### Motivation and Context

Fixes #14852

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

___

### **PR Type**
enhancement, tests

___

### **Description**
- Updated NUnit assertions in `DevToolsProfilerTest.cs` to ensure `StartTime`, `EndTime`, and `TimeDeltas` are non-zero, enhancing test robustness.
- Improved message formatting in `DevToolsTestFixture.cs` by using string interpolation for `Assert.Ignore`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>DevToolsProfilerTest.cs</strong><dd><code>Update NUnit assertions for profiler validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

dotnet/test/common/DevTools/DevToolsProfilerTest.cs

<li>Changed assertions for <code>StartTime</code> and <code>EndTime</code> to check for non-zero <br>values.<br> <li> Updated assertion for <code>TimeDeltas</code> to check for non-zero values.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14878/files#diff-ba006639f1cf86df74d37deefad1e81caff9147974af82a3ab237f7ffcecbf9e">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>DevToolsTestFixture.cs</strong><dd><code>Use string interpolation for ignore message</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

dotnet/test/common/DevTools/DevToolsTestFixture.cs

<li>Modified <code>Assert.Ignore</code> message formatting to use string interpolation.<br> <br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14878/files#diff-e20c83eb5accdd4bc009518726c9d665c019b4939de04febc3aa6845143c75f5">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: dotnet/test/common/DevTools/DevToolsProfilerTest.cs [hunk 1]
```diff
@@ -131,12 +131,12 @@ private void ValidateProfile(CurrentCdpVersion.Profiler.Profile profiler)
         {
             Assert.That(profiler, Is.Not.Null);
             Assert.That(profiler.Nodes, Is.Not.Null);
-            Assert.That(profiler.StartTime, Is.Not.Null);
-            Assert.That(profiler.EndTime, Is.Not.Null);
+            Assert.That(profiler.StartTime, Is.Not.Zero);
```
Comment (RenderMichael) [inline dotnet/test/common/DevTools/DevToolsProfilerTest.cs:134]: Modern NUnit comes with some analyzers which explained that `profiler.StartTime`, `profiler.EndTime`, and `delta` are value types (all doubles) so can never be null. These tests pass if we assert instead that they are `!= 0`

Comment (nvborisenko) [reply dotnet/test/common/DevTools/DevToolsProfilerTest.cs:134]: Not clear what zero means. If we are in context of datetimes, let's try to be specific.

Comment (RenderMichael) [reply dotnet/test/common/DevTools/DevToolsProfilerTest.cs:134]: So I took a deeper look at this, and although it's storing datetime values, it does so as unix timestamps (and therefore, numbers).

image
image

NUnit 4 introduces an `Is.(Not.)Default` assertion, which is all we're really testing here (making sure the field is populated), but we don't have access to that yet. Either way, I hope with this clarification we can accept `Is.Not.Zero` as a meaningful assertion. Let me know if there's something else we should do (maybe parse to DateTime from unit timestamp?)

Comment (nvborisenko) [reply dotnet/test/common/DevTools/DevToolsProfilerTest.cs:134]: Thanks.

**Review by nvborisenko — APPROVED** [review]

**Comment by nvborisenko** [comment]
Thank you @RenderMichael !

**Pull request merged**